repobee_sanitizer/sanitizer.py

```diff
@@ -82,10 +85,15 @@ def command(self) -> Optional[plug.Result]:
         else:
             LOGGER.info(f"Sanitizing repo and updating {self.target_branch}")
             try:
+                if self.create_pr_branch:
+                    pr_branch_name = "sanitizer-pull-request"
+                    _sanitize_repo.create_pr_branch(
+                        repo_root, self.target_branch, pr_branch_name
+                    )
+                    self.target_branch = pr_branch_name
```

Comment:
I strongly recommend that you to the highest degree possible treat variables, especially object attributes, as immutable. Reassigning the target branch here could lead to massive confusion down the line, and it's also not at all necessary. Here's a clearer way to solve it.

```suggestion
                effective_target_branch = self._resolve_effective_target_branch()
```

Where `_resolve_effective_target_branch()` simply does the above stuff of creating the PR branch if necessary and returning it, or just returning the target branch if not sanitizing to a PR branch. The you pass `effective_target_branch` to `sanitize_to_target_branch()`

Reply:
Good idea, that will make the code that is now essentially
```
if pr:
    this
else:
    that
```
much cleaner

Reply:
Yes, typically when I end up writing stuff like that, I refactor it into a separate function such that the branching paths become as isolated and easy to follow as possible. It's also typical that the branching paths are two variations of "the same thing", and so naming that "thing" with a function usually makes it easier to follow the code.